xio-core/src/main/java/com/xjeffrose/xio/http/PersistentProxyHandler.java

```diff
@@ -0,0 +1,55 @@
+package com.xjeffrose.xio.http;
+
+import com.google.common.hash.Funnels;
+import com.xjeffrose.xio.client.ClientConfig;
+import com.xjeffrose.xio.core.Constants;
+import com.xjeffrose.xio.core.SocketAddressHelper;
+import com.xjeffrose.xio.server.RendezvousHash;
+import io.netty.util.internal.PlatformDependent;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import lombok.val;
+
+public class PersistentProxyHandler extends ProxyHandler {
+  private final RendezvousHash<CharSequence> persistentProxyHasher;
+  private final Map<String, ClientConfig> clientConfigMap = PlatformDependent.newConcurrentHashMap();
+
+  public PersistentProxyHandler(
+    ClientFactory factory, ProxyRouteConfig config, SocketAddressHelper addressHelper) {
+    super(factory, config, addressHelper);
+    val clientConfigs = config.clientConfigs();
+    persistentProxyHasher =
+      buildHasher(clientConfigMap, clientConfigs.size());
+  }
+
+  private RendezvousHash<CharSequence> buildHasher(
+    Map<String, ClientConfig> configMap, int configSize) {
+    List<String> randomIdPool = new ArrayList<>();
+    val clientConfigs = config.clientConfigs();
+
+    // map each client config to a randomly-generated ID
+    for (int i = 0; i < configSize; i++) {
+      String id = UUID.randomUUID().toString();
+      randomIdPool.add(id);
+      configMap.put(id, clientConfigs.get(i));
+    }
+
+    return new RendezvousHash<>(Funnels.stringFunnel(Constants.DEFAULT_CHARSET), randomIdPool);
+  }
+
+  @Override
+  public ClientConfig getClientConfig(Request request) {
+    String rawXFF = request.headers().get(X_FORWARDED_FOR).toString();
+    String remoteAddress = rawXFF;
+    if (rawXFF.contains(",")) {
+      // extract originating address
+      remoteAddress = rawXFF.substring(0, rawXFF.indexOf(","));
+    } else if (rawXFF == null) {
+      remoteAddress = request.host();
```

Comment:
This isn't what you want. `Request.host` returns the host that the client is trying to connect to. HTTP/1.1 calls this a host header, h2 tried to clarify by referring to it as an authority. We should probably come up with a less confusing name as well.

The base class has some code for updating X_FORWARDED_FOR, I'm wondering if we should put similar logic in there to get the client address back out:
https://github.com/jguamie/xio/blob/384a49d334beeb5125f1186c8826946f860ab141/xio-core/src/main/java/com/xjeffrose/xio/http/ProxyHandler.java#L93-L104

Reply:
Ok, will update.